### PR TITLE
Add webui patch to allow filtering via regex

### LINKED ISSUE #2028: Gerätefilter usw. mit Regexpattern
### Describe the solution you'd like

Anstatt nur mit string matches zu filtern, wäre es hin und wieder ganz hilfreich, mittels Regex-Pattern zu filtern.

Hintergrund: Ich habe Aktoren und Thermostate die bis auf das Prefix gleich heißen. zB. Ak1_Raum1 und Th1_Raum1
Da die beiden mehr oder weniger zusammengehören, möchte ich nun zB hiermit filtern: `(Th|Ak)1.*`, mit genau diesen zwei Ergebnissen.

### Describe alternatives you've considered

Manuell filtern

### Is your feature request related to a problem?

Manuell filtern fühlt sich halt ein bisschen umständlich an :)

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by IT-VBFK**
Closes #2028

### Description

Issue: #2028 

Mit freundlicher Unterstützung von @jp112sdl ☺️

### Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Contributing checklist

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** and **LICENSE** document.
- [x] I fully agree to distribute my changes under Apache 2.0 license.

### CODE REVIEW

File: buildroot-external/patches/occu/0170-WebUI-RegexSearchInDeviceLists.patch [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+@@ -7882,11 +7882,12 @@
+     //var patternList = m_value.split("|");
+     text            = text.toLowerCase();
+     
+-    for (var i = 0, len = patternList.length; i < len; i++)
+-    {
+-      if (0 <= text.indexOf(patternList[i])) { return true; }
+-    }
+-    
++    //for (var i = 0, len = patternList.length; i < len; i++)
```
Comment (jens-maus) [inline buildroot-external/patches/occu/0170-WebUI-RegexSearchInDeviceLists.patch:12]: Commenting things just out is not appropriate. Please delete the lines instead that you think are definitly not required.

**Review by jens-maus — CHANGES_REQUESTED** [review]

File: buildroot-external/patches/occu/0170-WebUI-RegexSearchInDeviceLists.patch (hunk 1 above)
Comment (jens-maus) [reply buildroot-external/patches/occu/0170-WebUI-RegexSearchInDeviceLists.patch:12]: Not resolved, see Line #12

Comment (IT-VBFK) [reply buildroot-external/patches/occu/0170-WebUI-RegexSearchInDeviceLists.patch]: Sorry, my fault :)

Now it should be resolved

Comment (jp112sdl) [reply buildroot-external/patches/occu/0170-WebUI-RegexSearchInDeviceLists.patch]: Now, the original //conInfo... commented lines are removed, too?

Sometimes I use them for debugging

Comment (IT-VBFK) [reply buildroot-external/patches/occu/0170-WebUI-RegexSearchInDeviceLists.patch]: hmm.. I don't know that this comments were from the original occu 🙈 

I could re-add them if necessary...

File: buildroot-external/patches/occu/0170-WebUI-RegexSearchInDeviceLists.patch [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+       if (typeof obj['type'] != 'undefined') {
+         if (obj['type'] === "")                      { return false; }
+-        if (obj['type'].toLowerCase().indexOf(this.filType) == -1) { return false; }
++        var r = new RegExp(this.filType);                                           
++        if (r.test(obj['type'].toLowerCase()) === false) {return false; }
+       }
+     }
+     if (this.filDesc !== "")
+     {
+-      //conInfo( "iseFilter: filDesc="+this.filDesc );
```
Comment (jp112sdl) [inline buildroot-external/patches/occu/0170-WebUI-RegexSearchInDeviceLists.patch:28]: Keep this line

File: buildroot-external/patches/occu/0170-WebUI-RegexSearchInDeviceLists.patch [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+       if (typeof obj['desc'] != 'undefined')
+       {
+-        //conInfo( "iseFilter: desc="+obj['desc'] );
```
Comment (jp112sdl) [inline buildroot-external/patches/occu/0170-WebUI-RegexSearchInDeviceLists.patch:31]: Keep this line

**Comment by IT-VBFK** [comment]
Something went completely wrong here..

**Comment by jens-maus** [comment]
> Something went completely wrong here.. 
> 
> I think the issue here is the file encoding. So: what encoding this files must have? Obviously not UTF8...

Encoding and line endings have to be kept the same like in the original file from eQ3 even thought eq3 mixes them in different files themselves.

File: buildroot-external/patches/occu/0170-WebUI-RegexSearchInDeviceLists.patch [hunk 4]
```diff
@@ -0,0 +1,93 @@
+--- occu/WebUI/www/webui/webui.js.orig
++++ occu/WebUI/www/webui/webui.js
+@@ -7872,26 +7872,21 @@
+   };
+   
+   /**
+-   * Pr�ft, ob der Filter auf einen Text zutrifft
++   * Prüft, ob der Filter auf einen Text zutrifft
```
Comment (jens-maus) [inline buildroot-external/patches/occu/0170-WebUI-RegexSearchInDeviceLists.patch:8]: Please revert this unnecessary change.

File: buildroot-external/patches/occu/0170-WebUI-RegexSearchInDeviceLists.patch [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    **/
+   this.match = function(text)
+   {
+     if (false === isSet()) { return true; }
+     
+     var patternList = m_value.toLowerCase().split("|");
+-    //var patternList = m_value.split("|");
```
Comment (jens-maus) [inline buildroot-external/patches/occu/0170-WebUI-RegexSearchInDeviceLists.patch:15]: keep this line even if it is commented out.

File: buildroot-external/patches/occu/0170-WebUI-RegexSearchInDeviceLists.patch [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+-    }
+-    
++    var r = new RegExp(m_value.toLowerCase());                                                                                                                  
++    if (r.test(text) === true) {return true; } 
+     return false;
+   };
+   
+   /**
+-   * Schlie�t den Filter und ruft dei Callback-Funktion auf
++   * Schließt den Filter und ruft dei Callback-Funktion auf
```
Comment (jens-maus) [inline buildroot-external/patches/occu/0170-WebUI-RegexSearchInDeviceLists.patch:30]: unnecessary change.

**Review by jens-maus — CHANGES_REQUESTED** [review]

File: buildroot-external/patches/occu/0170-WebUI-RegexSearchInDeviceLists.patch [hunk 7]
```diff
@@ -0,0 +1,92 @@
+--- occu/WebUI/www/webui/webui.js.orig
++++ occu/WebUI/www/webui/webui.js
+@@ -7872,26 +7872,21 @@
+   };
+   
+   /**
+-   * Pr�ft, ob der Filter auf einen Text zutrifft
++   * Pr�ft, ob der Filter auf einen Text zutrifft
```
Comment (IT-VBFK) [inline buildroot-external/patches/occu/0170-WebUI-RegexSearchInDeviceLists.patch:8]: How do I revert such change?

**Review by jens-maus — APPROVED** [review]
